scripts/lib/CIME/XML/env_mach_pes.py

```diff
@@ -19,6 +19,32 @@ def __init__(self, case_root=None, infile="env_mach_pes.xml", components=None):
         schema = os.path.join(get_cime_root(), "config", "xml_schemas", "env_mach_pes.xsd")
         EnvBase.__init__(self, case_root, infile, schema=schema)
 
+    def set_value(self, vid, value, subgroup=None, ignore_type=False):
+        """
+        Set the value of an entry-id field to value
+        Returns the value or None if not found
+        subgroup is ignored in the general routine and applied in specific methods
+        """
+        vid, comp, iscompvar = self.check_if_comp_var(vid, None)
+        if vid == "COUPLER_COUNT":
+            if value > 1:
+                for othercomp in self._components:
+                    if othercomp != "CPL":
+                        ninst_string = "NINST_{}".format(othercomp)
+                        expect(self.get_value(ninst_string)==1,
+                               "Cannot change COUPLER_COUNT value if {} > 1".format(ninst_string))
+            elif value < 0:
+                # negative value effectively overrides safety check
+                value = -value
```

Comment:
Or you could fix the interface so that secret backdoors are not needed. Right now, I can execute `./xmlchange COUPLER_COUNT=-15` and it happily sets `COUPLER_COUNT`  to `15`. Since `xmlchange` is part of our user interface, this feels like a bad feature to me.